### ISSUE TITLE
Some cells are missing in page.find_tables()

### ISSUE THREAD

**Issue by asher-stern-aquant** [pymupdf/PyMuPDF#2892]
## Describe the bug (mandatory)
In some cases, some of the cells in a table are not included when extracting the table by `table.extract()`
The table has four cells in each row, but only two of them are included.

## To Reproduce (mandatory)
[27.pdf](https://github.com/pymupdf/PyMuPDF/files/13672030/27.pdf)

Run this code:
```python
import fitz

filename = '/Users/astern/Temporary/27.pdf'
with fitz.open(filename) as doc:
    for page in doc:
        table_finder = page.find_tables()
        for table in table_finder:
            for row in table.extract():
                print(row)
            print()
```

The result of the table extraction is:
```
['No.', 'Plain Text Designation', 'p/n', 'Label Designation']
['Power supply 960 Watt', '30043005']
['Screw set', '30043035']
['Side extension 300', '30060212']
['Upper deck segment profile', '30043030']
['Waste chute with lid', '30185036']
```

You can see that the first and last columns are included only in the header, but are not included in all other rows.

## Your configuration (mandatory)
 - Operating system: `macOS sonoma 14.1.2`
 - Python version: `3.10.13`
 - PyMuPDF version `1.23.7`

`print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` 
```
3.10.13 (main, Sep 11 2023, 08:39:02) [Clang 14.0.6 ] 
 darwin 
 
PyMuPDF 1.23.7: Python bindings for the MuPDF 1.23.7 library.
Version date: 2023-11-30 00:00:01.
Built for Python 3.10 on darwin (64-bit).
```

**Comment by JorjMcKie** [comment]
No table detection is perfect. Even AI-based algorithms cannot detect all tables.
The main problem is to find enough graphical information (lines, rectangles) to help detecting cell boundaries.
In your case we have two complications:
1. Information overload in the **_header_**: the cells have white background rectangles
2. Table left and right borders are missing

Taken together, these factors are confusing the detection algorithm.
I have developed experimental code that aims to cope with these things:
* Ignore white rectangles without borders
* Add border lines to clusters of connect drawings

I am getting these results:
```python
doc=fitz.open("27.pdf")
page=doc[0]
tabs=page.find_tables()
len(tabs.tables)
2
pprint(tabs[0].extract())  # this is the company logo top-right - must be excluded somehow
[['', '']]
pprint(tabs[1].extract())
[['No.', 'Plain Text Designation', 'p/n', 'Label Designation'],
 ['16', 'Power supply 960 Watt', '30043005', 'POWERSUPPLY 90-264VAC 24VDC 960W'],
 ['17', 'Screw set', '30043035', 'SET SCREWS SPARE FLUENT'],
 ['18', 'Side extension 300', '30060212', 'SIDE EXTENSION 300'],
 ['19', 'Upper deck segment profile', '30043030', 'PROFILE WORKTABLE'],
 ['20', 'Waste chute with lid', '30185036', 'CHUTE WASTE STRAIGHT WITH LID']]
```

**Comment by JorjMcKie** [comment]
If you are interested in experimenting with this mod, please let me know.

**Comment by asher-stern-aquant** [comment]
Hi @JorjMcKie .
Thank you.
Yes, I will be happy to experiment with this mode.
Many thanks.

**Comment by JorjMcKie** [comment]
Here is the zipped Python file. It contains all the table code.
To "install" it, go to your Python installation folder of fitz (or fitz_new) and copy table.py into it - replacing the same-named file there.
That's all.
[table.zip](https://github.com/pymupdf/PyMuPDF/files/13695612/table.zip)

**Comment by asher-stern-aquant** [comment]
Thank you so much @JorjMcKie !

**Comment by JorjMcKie** [comment]
it works?

**Comment by asher-stern-aquant** [comment]
Yes, it worked. Thank you.

**Comment by JorjMcKie** [comment]
Very glad to hear this!

**Comment by asher-stern-aquant** [comment]
Hi @JorjMcKie 
Unfortunately, the new code raises an exception on the attached file.
[5651_p2.pdf](https://github.com/pymupdf/PyMuPDF/files/13716247/5651_p2.pdf)

```python
filename = '/Users/astern/Temporary/5651_p2.pdf'
with fitz.open(filename) as doc:
    for page in doc:
        table_finder = page.find_tables()
```

The exception:
```
Traceback (most recent call last):
  File "/Users/astern/Documents/code/reseach-drafts/above/src/scop_package/learn/bug3.py", line 12, in <module>
    table_finder = page.find_tables()
  File "/Users/astern/miniconda3/envs/scop/lib/python3.10/site-packages/fitz/table.py", line 2056, in find_tables
    make_edges(page, clip=clip, tset=tset)  # create lines
  File "/Users/astern/miniconda3/envs/scop/lib/python3.10/site-packages/fitz/table.py", line 1977, in make_edges
    EDGES.append(line_to_edge(line_dict))
  File "/Users/astern/miniconda3/envs/scop/lib/python3.10/site-packages/fitz/table.py", line 502, in line_to_edge
    edge["orientation"] = "h" if (line["top"] == line["bottom"]) else "v"
KeyError: 'top'
```

The original code does not raise an exception on this file.

Many thanks in advance.

**Comment by JorjMcKie** [comment]
Thanks for the feedback - I will look into this.

**Comment by JorjMcKie** [comment]
This has been fixed with version 1.23.14.

**Comment by asher-stern-aquant** [comment]
Thank you.
